crates/ide-completion/src/completions/item_list/trait_impl.rs

```diff
@@ -210,7 +210,7 @@ fn add_function_impl(
                 ast::AssocItem::Fn(func) => func,
                 _ => unreachable!(),
             };
-
+            // TODO: need `function_decl` that unwraps future in the return type
```

Comment:
Help wanted here🙋

Reply:
You'll have to manually traverse the AST here and stitch the edit together, that is something like `transformed_fn.ret_type()` to fetch the return type, then you can match on that to check if its an `ImplTrait` one etc